### ISSUE TITLE
[Discussion] Thailand Reporting Lag Dashboard

### ISSUE THREAD

**Issue by samkanga** [onaio/reveal-frontend#1570]
A program manager would like to have access to the `reporting lag` table details. Below is a sample mock-up of the expected presentation.

![image](https://user-images.githubusercontent.com/5908630/116555228-cc733e00-a904-11eb-99dd-dd4dd9ea4230.png)

From the dashboard, the user should be able to:
- [ ] Filter by average reporting lag (time from a form being completed in the Android device to it being synced on the server) and average form collection length (duration) by plan, user, operational area and timeframe. The client is open to suggestions on the format for the average reporting lag, average form collection length and date last synced. NB: A user can select multiple plans from the `plan` filter, multiple users from `users` filter, and multiple locations from the `locations` filter. The filters should be govern by selected filters. E.g. If a plan filter is selected, the users filter would only list users from the selected plan.
- [ ] When user clicks on the filter by option, they should be able to select a specific/multiple user/operational area/timeframe. Time frames options to be custom and selection options for the last 1 day, last 7 days, last 14 days and last 28days
- [ ] User should be able to sort this table based on descending reporting lag, form version, date last synced

**Points of Discussion:**
- [ ] How do you calculate average reporting lag
- [ ] Average form collection length

**Comment by ciremusyoka** [comment]
@samkanga 
1. Please provide more information about the `Table` we will use to get the above aggregations from?
2. Am sceptical about the filter drop downs above. For instance on Thailand local we have more than 100 plans. Should we have a drop down with 100 items?
cc @moshthepitt

**Comment by ciremusyoka** [comment]
Apologies I did hit the `Close with comment` by mistake.

**Comment by moshthepitt** [comment]
> Should we have a drop down with 100 items?

Maybe some kind of search?

@samkanga is the above sufficient to add this as a new feature?  Do we need mockups/designs?  Do we need to get feedback from whoever is in charge of the Reveal product?

**Comment by samkanga** [comment]
@ciremusyoka I will do additional follow-up with the client on the drop down options and if we can explore doing a filter by user before the option to filter by plans/locations.

@moshthepitt I will reach out to @rowo to help review this.

**Comment by rowo** [comment]
@samkanga 
- Where does this show up in Reveal? Is it a new page within Monitor?
- What do you mean "explore doing a filter by user before the option to filter by plans/locations." Is that just the order of the filters from left to right?
- A search in the filter would be good.
- is there a table in Reveal with this many filters we are working off of? Another table I'm seeing is a bit different and have search, table options, and pagination.
- Overall this requirement have the feeling of something very client-led, but not user led. Kind of general trying to solve every potential use case without prioritizing anything specific.

**Comment by rowo** [comment]
@samkanga see rsponses below sorry for the delay

1. For where it exists — it would be good to understand how the user gets here. I don't think I know enough to recommend whether it goes into superset or monitor. Do you have a suggestion and reasoning on one or the other?
2. The order of the filters won't determine the order a user will use them. So I don't see it solving the problem you are trying to solve. We'll need to support a long list if the user chooses not to start with filtering by user.
3. I though Mosh was recommending to do a search within the drop down itself. I'm not sure what the best way is to do an overall search for information that isn't displayed in the table (since the table shown doesn't display plan/locations or user)
4. For this one, I meant is there a existing similar table functionality in wherever it will live (either superset or monitor)?
5. This doesn't seem like it took the best process in terms of collaborating towards a solution with the client, but if we're just trying to get it done, the main things to figure out is where it'll live (which I don't know enough about when and how and who this program manager will need this info — so it's up to you or the client whether the best place is superset or monitor), and just this new functionality to accommodate long drop downs (i.e. having a smart search on the drop down)

**Comment by samkanga** [comment]
Thank you @rowo for your thoughts on this.
1. We had done a table close to this for Zambia and we placed it on the Monitor page under an IRS Performance Menu.
2. This is noted.
3. I had a discussion with the client and they were open to having a search instead of a drop down. @moshthepitt please advice on the best way to achieve the search for this feature.
4. There is a similar table on the Zambia monitor page. See sample screenshot below that can be accessed through the link here.
![image](https://user-images.githubusercontent.com/5908630/118762448-e0b0b800-b87e-11eb-8f7a-5e12567ba56a.png)
5. This is noted. I propose we go with the Monitor page based on the Zambia work.

**Comment by rowo** [comment]
@samkanga Just a thought — a table search instead of drop downs gives the user less control over what shows up. Filters are structured since they limit to specific columns. If the client is okay with that, the only thing I'd suggest is you have to show that information in the table columns.. i.e. add the user, plans/locations as columns.

+1 monitor page. Thanks for the links to the other tables.

**Comment by samkanga** [comment]
Thanks @rowo for the additional thoughts.  For the search I was thinking more about a multiple selection search similar to what we have on the filters. See sample screenshot below:

**Comment by samkanga** [comment]
@ciremusyoka have a look at the comment above on the search feature.

**Comment by samkanga** [comment]
@rowo @moshthepitt we had additional discussions with CHAI and they have proposed moving the report lag to superset too, so all the usage analytics features are in one place. In addition, they shared a sample table created by Akros which is close to what they are looking for. See screenshot below. Also on the screenshot, the filters have been moved to the left. In the screenshot the filter box widget in superset has been used for the filters. I also confirmed that it is possible to do multiple selects with a filterbox.  I wanted to also check if you have any reservations to this approach.
![image](https://user-images.githubusercontent.com/5908630/119779371-46312400-bed1-11eb-9812-2e7bafe2a600.png)
